### PR TITLE
refactor: Rename Blockly.connectionTypes to Blockly.ConnectionType

### PR DESCRIPTION
**Pull request by himanshujaidka**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

#### Behavior Before Change
Blockly.connectionTypes

#### Behavior After Change
Blockly.ConnectionType

###Reason for Changes
Renamed Blockly.connectionTypes to Blockly.ConnectionType for consistency.

Additional context

issue linked #5156

### CODE REVIEW

File: core/block.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ const Tooltip = goog.require('Blockly.Tooltip');
 const VariableModel = goog.requireType('Blockly.VariableModel');
 /* eslint-disable-next-line no-unused-vars */
 const Workspace = goog.requireType('Blockly.Workspace');
-const connectionTypes = goog.require('Blockly.connectionTypes');
+const connectionTypes = goog.require('
```
Comment (cpcallen) [inline core/block.js:41]: 1. Something seems to have gone wrong here.  Failed search-and-replace?
2. The local variable should be renamed too—if the module is `Blockly.ConnectionType` then the local alias should be `ConnectionType`.

Here in several other files.

File: core/blockly.js [hunk 2]
```diff
@@ -402,24 +402,24 @@ Blockly.ALIGN_RIGHT = Blockly.constants.ALIGN.RIGHT;
  */
 
 /**
- * @see Blockly.connectionTypes.INPUT_VALUE
+ * @see 
  */
-Blockly.INPUT_VALUE = Blockly.connectionTypes.INPUT_VALUE;
+Blockly.INPUT_VALUE =
```
Comment (cpcallen) [inline core/blockly.js:407]: Failed search-and-replace?  Here and 3x below.

**Review by cpcallen — CHANGES_REQUESTED** [review]
A few general remarks:

* A thumbs up to this work landing on `goog_module`, when it's ready.
* Please don't commit a rebuilt `blockly_compressed.js` (or any other compiled code, including `typings/blockly.d.ts`) to this branch.
* Do make sure to `npm run build:deps` to fix `tests/deps.js` after fixing the other issues outlined below.

One additional thing: we will need to keep a record of renames like this where they affect the public API (there will be many more soon, as a result of work on the `goog.module` transition and other cleanup).  I have a plan about how to do so, but this PR should not be committed until that plan has been implemented.

File: core/blockly.js (hunk 2 above)
Comment (himanshujaidka) [reply core/blockly.js:407]: 😅yes I think here search-and-replace is failing, okay sir I will look into this

**Comment by himanshujaidka** [comment]
Screenshot (75)
@cpcallen sir all tests are passed but now there are conflicts in deps.js file so how to manage that, pls help me in that

File: core/blockly.js [hunk 3]
```diff
@@ -412,24 +412,24 @@ Blockly.ALIGN_RIGHT = Blockly.constants.ALIGN.RIGHT;
  */
 
 /**
- * @see Blockly.connectionTypes.INPUT_VALUE
+ * @see Blockly.ConnectionTypes.INPUT_VALUE
```
Comment (cpcallen) [inline core/blockly.js:415]: Here is one example of a lint error that needs fixing.

File: core/blockly.js [hunk 4]
```diff
@@ -412,24 +412,24 @@ Blockly.ALIGN_RIGHT = Blockly.constants.ALIGN.RIGHT;
...
 /**
- * @see Blockly.connectionTypes.INPUT_VALUE
+ * @see Blockly.ConnectionTypes.INPUT_VALUE 
  */
-Blockly.INPUT_VALUE = Blockly.connectionTypes.INPUT_VALUE;
+Blockly.INPUT_VALUE = Blockly.ConnectionTypes.INPUT_VALUE;
 
 /**
- * @see Blockly.connectionTypes.OUTPUT_VALUE
+ *  @see Blockly.ConnectionTypes.INPUT_VALUE
```
Comment (cpcallen) [inline core/blockly.js:420]: This line has extraneous leading and trailing spaces.

**Review by cpcallen — CHANGES_REQUESTED** [review]
@himanshujaidka: I've added three commits (one a merge to bring the branch up-to-date compared to `goog_module`, one to un-revert `package-lock.json`, and one to add an example entry to `renamings.js`).

I note that the title of the PR is "Renamed Blockly.connectionTypes to Blockly.ConnectionType", but in fact you've renamed it to `ConnectionTypes` (plural).  I think the singular name is preferable (and I've used it in the entry I propose for `renamings.js`.  Can you update the branch to reflect the title?

Also: there are are some lint errors that need correction.

Once everything is done, please make sure that `npm test` completes successfully and then push your changes to this branch for final review.

**Comment by himanshujaidka** [comment]
@cpcallen sorry sir for the late reply, sure I will look into this

**Comment by himanshujaidka** [comment]
Screenshot (229)
Now in `test/deps.js` some lines are showing error, but it is less as compared to the previous version

**Review by cpcallen — APPROVED** [review]
I've done another merge from `goog_module` to this branch (and fixed a few small issues so tests pass) and added one final commit to rename `core/connection_types.js` to `connection_type.js`.

Nice work.  Thanks for your help!

**Comment by cpcallen** [comment]
Closes #5156.

**Comment by himanshujaidka** [comment]
> Nice work. Thanks for your help!

Thanks for your support

**Pull request merged**